### ISSUE TITLE
Navigation Sub Menu: Bugs in color handling and block context

### ISSUE THREAD

**Issue by yogeshbhutkar** [WordPress/gutenberg#68350]
### Description

Here are the issues identified while interacting with the Navigation Submenu component:

1. **Color Panel Behavior**:  
   The color panel for the Navigation Submenu component allows setting text and background colors. However, these colors only work with predefined values sourced from `template/theme.json`. Custom values are not applied successfully.

![Image](https://github.com/user-attachments/assets/c93df2b1-1085-41ea-ab21-eba07f0c3e43)

![Image](https://github.com/user-attachments/assets/2702cda9-051c-4322-acbe-20d98cc2a52a)

2. **Color Application Discrepancy**:  
   When text and background colors are set for the submenu component, they correctly apply to the submenu itself, leaving the overlay text and background settings unaffected in the editor (as intended). However, on the front end, these colors are mistakenly applied to the submenu container instead of the submenu parent, overriding the overlay settings and causing inconsistency between the `Editor` and the `Front End`.

![Image](https://github.com/user-attachments/assets/5d8a1835-9641-4bca-ac5a-2270f3620eff)

I'm unsure if these issues are a regression resulting from recent changes to the component.

### Step-by-step reproduction instructions

1. Navigate to a `Post/Page` edit screen.
2. Create a `Navigation Block`.
3. Try setting up different `permutations` for `text` and `background` properties along with `custom colors`.
4. Notice the discrepancies stated above.

### Screenshots, screen recording, code snippet

Corresponding screenshots and recordings are provided in the issue description.

### Environment info

- Gutenberg Plugin: Trunk (5af740f084d1f8bdd0fcb9724a2dbb0daf8a9213)
- WordPress Version: 6.8-alpha-59361
- OS: macOS Sequoia

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by carolinan** [comment]
I am not 100% sure about this comment:
> if only submenu's color and background value are passed, then it shouldn't override the value of the submenu container.

I don't think that this is expected by users, because there is an issue reported where the user is unable to change the colors when styling the submenu **block**.

https://core.trac.wordpress.org/ticket/62739

**Comment by yogeshbhutkar** [comment]
I just reviewed the ticket, I think, the user is trying to change the color in the following manner:
![Image](https://github.com/user-attachments/assets/7f05ef29-5a90-4dea-99da-2ec49df2ecc5)


But I've been doing it like this and this should work (even on the trunk).
![Image](https://github.com/user-attachments/assets/f0b7fb30-d01e-4be8-8a49-777d5226bd92)

Applying colors using the styles tab -> block -> submenu isn't working. But, it's possible to add colors to the `submenu` block from the `block` panel in the sidebar (demonstrated in second screencast).

I'll try digging a bit deeper on why exactly the global block styles aren't working for the `submenu`.

CC: @carolinan

**Comment by carolinan** [comment]
I don't think it is clear at all if a user wants to change the colors of the **parent** of the submenu, the submenu dropdown container, or, the menu items in the submenu. Changing one option may change the other uses cases.

It is one of these issues that would require a refactor of the navigation block.
